metals/src/main/scala/scala/meta/internal/metals/ScalacDiagnostic.scala

```diff
@@ -30,4 +30,22 @@ object ScalacDiagnostic {
         case _ => None
       }
   }
+
+  object SymbolNotFoundWithImportSuggestions {
+    // https://github.com/lampepfl/dotty/blob/3b741d67f8631487aa553c52e03ac21157e68563/compiler/src/dotty/tools/dotc/typer/ImportSuggestions.scala#L311-L344
+    private val regexes = for {
+      fix <- List("One of the following imports", "The following import")
+      help <- List("fix ", "make progress towards fixing")
+    } yield s"""(?s)$fix imports might $help the problem:(.*)""".r
```

Comment:
The exact format of the compiler message strings is not stable and may change at any time, please do not rely on it. Instead, I encoruage you to contribute upstream a way to expose this information semantically in our Reporting API (we also implement scalafix-like rewrites which could be displayed as quick fix actions in vscode).

Reply:
Revisiting long after the fact. @smarter has any thought gone into what this would look like, exposing this information semantically? I've been thinking about this often lately and sort of year for an error index like [rust has](https://doc.rust-lang.org/error-index.html) and more structured errors that we can use for stuff like this. I see some convos that touch on this a bit like [this old convo](https://contributors.scala-lang.org/t/towards-better-error-messages-in-scalac/1470/32), but has there been any design though or conversation around what this could look like in Dotty?

Reply:
> has there been any design though or conversation around what this could look like in Dotty?

No, this is still waiting for a motivated individual to step up :).